### PR TITLE
Slimes and things

### PR DESCRIPTION
**Pull request by Neerti**
Ports slimes to new system, unchecks and removes old slimes, and maps NS/SC maps compile again.

Changes:
- Slimes are now simple_mobs and not simple_animals.
  - Slime damage has roughly been reduced in half to compensate for attacking twice as fast.
  - Probability for stun success has been reduced due to more disarm attempts being made.
- Mobs driven with the new AI will refuse to step onto an 'unsafe' turf.
  - Non-player mobs can now suffer from confusion, if their mob is confused. They will walk around randomly (similar to confused players) and may attack someone randomly if they bump into them. They will also walk onto unsafe tiles such as open space while not able to fly, or water tiles as a slime.
    - 'Smart' AIs will instead stand still.
- Lightning effects made more generic.
- Lightning on simple_mobs can be resisted with sufficently high ``shock_resist``.
- Fixes bug where modifiers made attack speed permanently faster (or slower) on simple_mobs.
- Fixes health bug where a simplemob's max health was being set to their current health on initialization instead of the other way around.
- AI will not try to fight ``/mob/z_shadow``s now.
- Feral slimes are now their own type, under a parent slime class instead of being a subclass of a specific color. The feral slime type is more lean in terms of code and does not include things like discipline/reproduction/etc.
  - Dark blue feral slime now has a deadly icicle attack that fires every three seconds, with a half second windup time. Its aura also applies a short term Chilled modifier in addition to its direct damage, which slows movement, makes attacking slower, and makes disabling effects last longer. It is not applied if 100% cold immune.
- Slimebatons and xenotasers work on simple_mobs with the ``MOB_CLASS_SLIME`` flag, which will in the future include blob monsters.
  - Feral slimes hit with a slimebaton/xenotaser are taunted to the attacker.
- The orange, dark blue, and yellow slimes have reduced attack strength but inflict additional bonus 'elemental' damage, associated with their color.
- Certain simple_mobs such as ice spiders, thermic spiders, orange/blue/dark blue/yellow slimes, etc, are resistant or immune to their associated 'element'. Some of these have received a vulnerability towards another 'element'.
- Dark blue cold aura does direct damage instead of setting temperature, meaning it will give more time for a xenobiologist to react before the other slimes die. The aura also counts as a hostile attack to other mobs caught inside that are not immune to cold.
- Slimes can be ordered to squish if sufficiently obedient/friendly/pacified.
- Cleans up a lot of code involving sleepers/cryo/mech/etc entry that made sure you couldn't get in with a slime eatting you. It now checks for buckled mobs, which is both cleaner and futureproofs against future mobs that can buckle themselves to you.
- A cosmetic 'consumption beam' is shown when a slime absorbs material from another non-robotic mob when attacking.
- Probably other stuff I'm forgetting that I'll add to this list later.

### CODE REVIEW

File: code/modules/mob/living/simple_mob/subtypes/slime/slime.dm [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+			I.appearance_flags = RESET_COLOR
+			MA.overlays += I
+
+		// Mood overlay.
+		I = image(icon, src, "aslime-[mood]")
+		I.appearance_flags = RESET_COLOR
+		MA.overlays += I
+
+	// Hat simulator.
+	if(hat)
```
[the remark this answers is not in the document]
Comment (Neerti) [reply code/modules/mob/living/simple_mob/subtypes/slime/slime.dm]: Feral slimes are a subtype of the base slime type, so hats should work on them.

File: code/modules/mob/living/simple_mob/subtypes/slime/xenobio/consumption.dm [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+		amount_grown = between(0, amount_grown + 1, 10)
+
+// Called if above proc happens while below a nutrition threshold.
+/mob/living/simple_mob/slime/xenobio/proc/handle_starvation()
+	if(nutrition < get_starve_nutrition() && !client) // if a slime is starving, it starts losing its friends
+		if(friends.len && prob(1))
+			var/mob/nofriend = pick(friends)
+			if(nofriend)
+				friends -= nofriend
+				say("[nofriend]... food now...")
```
[the remark this answers is not in the document]
Comment (Neerti) [reply code/modules/mob/living/simple_mob/subtypes/slime/xenobio/consumption.dm]: This is what the old code had.

File: code/modules/mob/living/simple_mob/subtypes/slime/xenobio/subtypes.dm [hunk 3]
```diff
@@ -0,0 +1,783 @@
...
+		L.adjust_fire_stacks(1)
+		if(prob(12))
+			L.IgniteMob()
+
+/mob/living/simple_mob/slime/xenobio/blue
+	desc = "This slime produces 'cryotoxin' and uses it against their foes.  Very deadly to other slimes."
+	color = "#19FFFF"
+	slime_color = "blue"
+	coretype = /obj/item/slime_extract/blue
+	reagent_injected = "cryotoxin"
```
[the remark this answers is not in the document]
Comment (Neerti) [reply code/modules/mob/living/simple_mob/subtypes/slime/xenobio/subtypes.dm]: The old code used cryotoxin.

File: code/modules/mob/living/simple_mob/subtypes/slime/xenobio/subtypes.dm [hunk 4]
```diff
@@ -0,0 +1,783 @@
...
+/mob/living/simple_mob/slime/xenobio/dark_purple/ex_act(severity)
+	log_and_message_admins("[src] ignited due to a chain reaction with an explosion.")
+	ignite()
+
+/mob/living/simple_mob/slime/xenobio/dark_purple/fire_act(datum/gas_mixture/air, temperature, volume)
+	log_and_message_admins("[src] ignited due to exposure to fire.")
+	ignite()
+
+/mob/living/simple_mob/slime/xenobio/dark_purple/bullet_act(var/obj/item/projectile/P, var/def_zone)
+	if(P.damage_type && P.damage_type == BURN && P.damage) // Most bullets won't trigger the explosion, as a mercy towards Security.
```
[the remark this answers is not in the document]
Comment (Neerti) [reply code/modules/mob/living/simple_mob/subtypes/slime/xenobio/subtypes.dm]: Its what the previous code had, and lasers are generally the default lethal response so making lasers safe basically makes it not matter at all.

File: code/modules/mob/living/simple_mob/subtypes/slime/slime.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Anewbe) [reply code/modules/mob/living/simple_mob/subtypes/slime/slime.dm]: Okay, then the followup: Does the offset work for them, too?

File: code/modules/mob/living/simple_mob/subtypes/slime/xenobio/consumption.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (Anewbe) [reply code/modules/mob/living/simple_mob/subtypes/slime/xenobio/consumption.dm]: Strange, carry on them.

**Comment by Anewbe** [comment]
Fuck it, code seems fine, merging.

**Pull request merged**